### DISCUSSION TITLE
Use @canany not working when permission getting from database

### DISCUSSION THREAD

**Discussion by rizkhal** [spatie/laravel-permission#1759] [Q&A]
Hello, I want to use `@canany` blade directive to show or hidden some menu on view

My problem is when i using `@canany` and getting all permission from database like below, it's not working
```php
@canany(\App\Entities\Permission::get()->pluck('name')->toArray(), Auth::user())
    <li class="menu-header">Some menu</li>
@endcanany
```
but when the permission i write manual `['first_permission', 'second_permission', ...]` it's working

Where's my mistake?

**Comment by drbyte** [comment]
The `@canany()` Blade directive is [provided by Laravel](https://laravel.com/docs/master/authorization#gates-supplying-additional-context), not by this package.

You said that providing a hand-written array works, but calling your override model does not. So, what is the output of your `toArray()` command? And, why are you passing `, Auth::user()` to `@canany()`? Perhaps that's your error?

**Reply by rizkhal** [reply]
hello @drbyte thanks for you reply

my code is here https://github.com/codegoen/spp-sekolahala/blob/d82241762cc87a3c7a26c49c40ca5c126f1a5b24/resources/views/layouts/includes/sidebar.blade.php#L17

result of `permission` `toArray` same as array on the link above.

and i use `@canany` because on this docs https://spatie.be/docs/laravel-permission/v3/basic-usage/blade-directives#permissions say support `@canany` like `@can`

i dont have error message

**Reply by drbyte** [reply]
> result of `permission` `toArray` same as array on the link above.

Well, if they're the same then you'd be getting the same results. So they must not actually be the same.

What's the output of each of these, inside the Blade file?
```blade
@php 
$permissions_array = \App\Entities\Permission::get()->pluck('name')->toArray();
dump($permissions_array);
@endphp
@canany($permissions_array)
first
@endcanany
@canany($permissions_array, Auth::user())
second
@endcanany
```

```blade
@canany(\App\Entities\Permission::get()->pluck('name')->toArray(), Auth::user())
third
@endcanany
@canany(\App\Entities\Permission::get()->pluck('name')->toArray())
fourth
@endcanany
```

**Reply by rizkhal** [reply]
Sorry for my late reply.

Below is the output.

![Screenshot from 2021-06-21 16-27-34](https://user-images.githubusercontent.com/24653114/122731279-b0a96a00-d2ad-11eb-8788-122c8d71d8c4.png)
![Screenshot from 2021-06-21 16-27-45](https://user-images.githubusercontent.com/24653114/122731277-af783d00-d2ad-11eb-8921-52b8c182c1c1.png)

When i hard write everything is fine, whats wrong with?

**Reply by drbyte** [reply]
I don't understand. 
It shows ALL of the tests working properly: `first`, `second`, `third`, `fourth`.
And then your hand-written array shows nothing (no `Data Master`) (which is only half of the array).
